talisker/util.py

```diff
@@ -35,3 +38,60 @@ def parse_url(url, proto='http'):
 
 def pkg_is_installed(name):
     return name in [x.project_name for x in pip.get_installed_distributions()]
+
+
+# a module level cache for global objects
+_global_cache = {}
+_global_dicts = []
+_context_locals = []
+
+
+def module_cache(func):
```

Comment:
Nope, sorry! Take `get` as an example:

1. Thread 1 executes `if id not in _global_cache`, sees that there's nothing in the cache, but is then interrupted.
2. Thread 2 jumps in, also sees nothing in the cache, allocates a new instance, o2, adds it to the cache, and returns o2 to the caller.
3. Thread 1 wakes up again and continues by allocating a new instance, o1, adding it to the cache, and returning o1 to the caller.

You now have one client with o1, one with o2, and a cache containing o1 (although it could just as easily be o2 in the cache, with o1 as the "floating" instance). Next time a client calls in, it will get the cached version but the service may be inconsistent by then.

Note: there are other points where an interrupt could occur, with similar effects.

But, like I said, maybe it doesn't actually matter for the current use case.